go.mod

```diff
@@ -4,75 +4,68 @@ go 1.12
 
 require (
 	github.com/container-storage-interface/spec v1.1.0
-	github.com/golang/groupcache v0.0.0-20190129154638-5b532d6fd5ef // indirect
 	github.com/golang/mock v1.2.0
-	github.com/google/go-cmp v0.3.1 // indirect
-	github.com/googleapis/gnostic v0.2.0 // indirect
-	github.com/imdario/mergo v0.3.7 // indirect
-	github.com/kubernetes-csi/csi-lib-utils v0.6.1
+	github.com/kubernetes-csi/csi-lib-utils v0.6.2-0.20191223231529-62bc1d7a544e
 	github.com/kubernetes-csi/csi-test v2.0.0+incompatible
-	github.com/kubernetes-csi/external-snapshotter v2.0.0-rc1.0.20191024235139-bc6e42db5bc6+incompatible
+	github.com/kubernetes-csi/external-snapshotter v1.2.1-0.20191220180133-bba358438aee
 	github.com/miekg/dns v1.1.8 // indirect
-	github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90 // indirect
-	github.com/prometheus/common v0.2.0 // indirect
-	github.com/prometheus/procfs v0.0.0-20190403104016-ea9eea638872 // indirect
-	github.com/spf13/pflag v1.0.3
-	golang.org/x/time v0.0.0-20190308202827-9d24e82272b4 // indirect
-	google.golang.org/grpc v1.19.1
-	gopkg.in/inf.v0 v0.9.1 // indirect
-	k8s.io/api v0.0.0-20191005115622-2e41325d9e4b
-	k8s.io/apimachinery v0.0.0-20191006235458-f9f2f3f8ab02
-	k8s.io/apiserver v0.0.0-20190918200908-1e17798da8c1
-	k8s.io/client-go v0.0.0-20190918200256-06eb1244587a
-	k8s.io/component-base v0.0.0-20190918200425-ed2f0867c778
-	k8s.io/csi-translation-lib v0.0.0-20191009030015-17db17aaadeb
+	github.com/spf13/pflag v1.0.5
+	google.golang.org/grpc v1.26.0
+	k8s.io/api v0.17.0
+	k8s.io/apimachinery v0.17.1-beta.0
+	k8s.io/apiserver v0.17.0
+	k8s.io/client-go v0.17.0
+	k8s.io/component-base v0.17.0
+	k8s.io/csi-translation-lib v0.17.0
 	k8s.io/klog v1.0.0
 	k8s.io/kubernetes v1.14.0
 	sigs.k8s.io/sig-storage-lib-external-provisioner v4.0.1+incompatible
 )
 
-replace k8s.io/api => k8s.io/api v0.0.0-20190918195907-bd6ac527cfd2
+replace k8s.io/api => k8s.io/api v0.17.0
```

Comment:
Were these forced overrides or automatic?  I'm wondering if we can actually remove most of these replace statements now that most of the repos have switched to semantic versioning.

Reply:
Automatic via go-get-kubernetes.sh
I'll do some clean up.

Reply:
> I'm wondering if we can actually remove most of these replace statements now that most of the repos have switched to semantic versioning.

We can't, because the semantic versioning for Kubernetes components is too weak: `go get -u ./...` for updating to the latest "minor and patch upgrades" of other dependencies would also update all Kubernetes components if we removed the replace statements. This is bad because there's no guarantee that 0.18.0 is really API compatible with 0.17.0. I think it's better to update Kubernetes via the script and use `go get -u ./...` for the rest.